### PR TITLE
Forward select Rectangle kwargs to Connection patch

### LINKED ISSUE #23424: [Bug]: Axes.indicate_inset[_zoom](linewidth=...) doesn't affect connectors
### Bug summary

When using `Axes.indicate_inset`/`Axes.indicate_inset_zoom`, it may be necessary to increase the size of edges in order to be visible. It is possible to change edge colour with `edgecolor=...` which affects both the frame and the connectors. However, using `linewidth=...` only affects the frame.

### Code for reproduction

```python
import matplotlib.pyplot as plt

fig, axs = plt.subplots(2)
for ax in axs:
    ax.plot([1, 2, 3])

axs[1].set_xlim(0.5, 1.5)
axs[0].indicate_inset_zoom(axs[1], linewidth=5)

plt.show()
```
```


### Actual outcome

![Figure_1](https://user-images.githubusercontent.com/302469/178911771-21e2f05b-4b23-4678-8f83-9369f40ca83b.png)

### Expected outcome

![Figure_1](https://user-images.githubusercontent.com/302469/178912221-a0571a94-350b-452f-b960-bedc2365cd52.png)

### Additional information

_No response_

### Operating system

_No response_

### Matplotlib Version

37ccdca5028725b3b9d689b42947b6284a73d690

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

git checkout

### PR DESCRIPTION
**Pull request by ejmercier**
## PR Summary
Closes #23424
This bug fix forwards `**kwargs` from the indicate_inset function into the `Connection Patch`. It uses a blacklist to decide which ones should be passed as `patch_kwargs` to the `Connection Patch` and not the `Rectangle patch`.
## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A ] New features are documented, with examples if plot related.
- [N/A ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by oscargus** [comment]
Thanks! Can you please add a test? 

I think a suitable test is to set some of the parameters and then check that the resulting ConnectorPatch have the expected values.

**Comment by ejmercier** [comment]
Yes I can add a test. Is there a specific place where I should create it?

**Comment by oscargus** [comment]
I think that `test_axes.py` would be a good place. https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/tests/test_axes.py

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -474,11 +474,22 @@ def indicate_inset(self, bounds, inset_ax=None, *, transform=None,
                 if self.yaxis.get_inverted():
                     ey = 1 - ey
                 xy_data = x + ex * width, y + ey * height
+
+                blacklist = {
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:478]: We already use `allowed_keys` elsewhere for this:
```suggestion
                disallowed_keys = {
```

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -474,11 +474,22 @@ def indicate_inset(self, bounds, inset_ax=None, *, transform=None,
...
+                    'alpha', 'rasterized', 'transform',
+                    'color', 'facecolor', 'url',
+                    'hatch', 'in_layout', 'path_effects'}
+
+                patch_kwargs = {
+                    key: val for key, val in kwargs.items()
+                    if key not in blacklist}
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:485]:
```suggestion
                    if key not in disallowed_keys}
```

**Comment by rcomer** [comment]
Hi @ejmercier are you still interested in working on this one?  It looks like it still needs a test.

I'm going to mark this as "draft" for now.  Please mark it as "ready for review" when you are ready.  In the meantime, if you need help, feel free to ask questions here.  Or you may prefer to ask them in our [incubator gitter room](https://matplotlib.org/stable/devel/contributing.html#contributor-incubator).

**Comment by ejmercier** [comment]
@rcomer thanks for checking up! I'm still looking to work on this. I'll plan to write the test soon and move it out of drafts! If I have any questions I'll reach out

**Pull request closed without merging**